mobile-app/app/screens/AppNavigator/screens/Loans/hooks/CollateralPrice.ts

```diff
@@ -58,3 +46,51 @@ export function useResultingCollateralRatio (collateralValue: BigNumber, existin
     new BigNumber(existingLoanValue).plus(newLoanAmount.multipliedBy(interestPerBlock.plus(1)).multipliedBy(
       activePrice))).multipliedBy(100)
 }
+
+export function getCollateralPrice (amount: BigNumber, collateralItem: CollateralItem | CollateralToken, totalCollateralValue: BigNumber): CollateralPrice {
+  const activePrice = new BigNumber(getActivePrice(collateralItem.token.symbol, collateralItem.activePrice))
+  const collateralPrice = activePrice.multipliedBy(amount)
+  const collateralFactor = new BigNumber(collateralItem?.factor ?? 0)
+  const vaultShare = getVaultShare(amount, collateralFactor, activePrice, new BigNumber(totalCollateralValue.isZero() ? collateralPrice : totalCollateralValue))
+  return {
+    activePrice,
+    collateralPrice,
+    vaultShare,
+    collateralFactor
+  }
+}
+
+export function getVaultShare (collateralAmount: BigNumber, factor: BigNumber, price: BigNumber, totalCollateralValue: BigNumber): BigNumber {
+  const vaultShare = new BigNumber(collateralAmount).multipliedBy(price).multipliedBy(factor).dividedBy(totalCollateralValue)
+  return BigNumber.max(BigNumber.min(1, vaultShare), 0).times(100)
+}
+
+export function useValidCollateralRatio (
+  collateralAmounts: LoanVaultTokenAmount[],
+  totalCollateralVaultValue: BigNumber,
+  collateralTokenId?: string,
+  updatedCollateralAmount?: BigNumber
+): {requiredVaultShareTokens: string[], isValidCollateralRatio: boolean, requiredTokensShare: BigNumber} {
+  const requiredVaultShareTokens = ['DUSD', 'DFI']
+  const collateralTokens = useSelector((state: RootState) => state.loans.collateralTokens)
+  const requiredTokensShare = requiredVaultShareTokens.reduce((share, tokenSymbol) => {
+    const collateralItem = collateralTokens.find((col) => col.token.symbol === tokenSymbol)
+    if (collateralItem !== undefined) {
+      const collateralToken = collateralAmounts.find((t) => t.symbol === tokenSymbol) ?? {
+        ...collateralItem?.token,
+        amount: '0'
+      }
+      const amount = collateralToken.id === collateralTokenId ? updatedCollateralAmount ?? 0 : new BigNumber(collateralToken.amount)
```

Comment:
Is amount inconsistent? You use ?? 0 but the else is bignumber

Reply:
Will update the logic